### ISSUE TITLE
A silent crash in Id-statement

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#31]
The following program crashes without any useful messages in the Id-statement (actually no match with F):

```
#-
* on Linux x86_64
#:maxtermsize 40000
#:workspace 40000000
#define N "700"
#define M "200"
CF Z,f,acc;
S n1,...,n`N';
S a1,...,a`N';
L F = Z(0) * f(<acc(n1,a1)>,...,<acc(n`N',a`N')>);
*L G = Z(0) * f(<acc(n1,a1)>,...,<acc(n`N',a`N')>,acc(1,1));
* Comsume the workspace.
#do i=1,`M'
  id Z(?a) = Z(?a);
#enddo
id f(?a,acc(a1?,1),?b) = f(acc(a1,1),?a,?b);
.end
```

```
FORM 4.1 (Apr 24 2015) 64-bits                   Run: Fri Jun  5 15:47:24 2015
    #-
Program terminating at foo.frm Line 16 --> 
  0.03 sec out of 0.03 sec
```

It looks like due to a shortage of the workspace, because setting M=100 makes the program work well.

**Comment by tueda** [comment]
Another silent crash (on Linux x86_64). Maybe the origin could be different from the above case because `valgrind` detects "Invalid read of size 4" in `Normalize()`.

https://gist.githubusercontent.com/tueda/4430887c7e34fcd25c40/raw/53ecf0a26c95b3e293e2e9ba31a834cb43af5e7e/20150630.frm

**Comment by vermaseren** [comment]
I had a quick look.
Obviously it ran out of some buffer. The statement where it did crash (when you turn on MALLOCPROTECT) is a rather normal statement and the addresses looked reasonable as well.
It will be rather difficult to debug this, because having almost 10000 arguments makes it a bit
hard to track the function and to inspect its binary representation.
When I run it regularly in the debugger it gets through......

**Comment by tueda** [comment]
It seems that this issue was fixed in a8b0f2d37b24fc528ab87c05ad62713faf0a5cc6.

```
GOOD * 66a04da [2016-04-06] [test] Check if test/unit is loaded @Takahiro Ueda
GOOD * 7252064 [2016-03-29] Compiler error when the same label is defined more than once @Takahiro Ueda
GOOD * a8b0f2d [2016-02-18] Added errormessage for workspace overflow in MatchFunction. @vermaseren
BAD  *   98cfbfb [2016-02-18] Needs to get Takahiros update first... Merge branch 'master' of github.com:vermaseren/form @vermaseren
     |\
BAD  | * e9643be [2016-02-16] Fix ToTensor !<vector or set> @Takahiro Ueda
BAD  * | 07edf53 [2016-02-18] Fixed a bug in the transform statement that did not take fixed args @vermaseren
     |/
BAD  * fa74cbf [2016-02-12] Repaired a bug with combining bracketindex and intohide. @vermaseren
```

**Comment by tueda** [comment]
The ["another silent crash"](https://github.com/vermaseren/form/issues/31#issuecomment-117117082) seems to be a duplicate of #268 (there are 9486 arguments matched with `?a`) and will be fixed by #490.
